### PR TITLE
fedora: add SELinux validation test suite

### PR DESCRIPTION
**Pull request by balakreddy**
## Description

Add selinux_validation.py with two test cases for Fedora cloud images: verify_setenforce and verify_selinux_info. Covers mode toggling, audit log verification (MAC_STATUS records), sestatus/avcstat field checks, and edge cases (selinuxfs unmounted, enforce file bind-mounted RO).

Sources:
https://src.fedoraproject.org/tests/selinux/blob/main/f/libselinux/setenforce https://src.fedoraproject.org/tests/selinux/blob/main/f/policycoreutils/selinux-info

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation update

## Checklist

- [x] Description is filled in above
- [x] No credentials, secrets, or internal details are included
- [x] Peer review requested (if not, add required peer reviewers after raising PR)
- [x] Tests executed and results posted below

## Test Validation

**Key Test Cases:**

verify_selinux_info|verify_setenforce

**Impacted LISA Features:**

**Tested Azure Marketplace Images:**

- WestUS3/Fedora-5e266ba4-2250-406d-adad-5d73860d958f/Fedora-Cloud-44-x64/latest

## Test Results

| Image | VM Size | Result |
|-------|---------|--------|
| WestUS3/Fedora-5e266ba4-2250-406d-adad-5d73860d958f/Fedora-Cloud-44-x64/latest      | Standard_D2ads_v5        | PASSED |

2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner ________________________________________
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner          FedoraSELinuxValidation.verify_setenforce: PASSED   
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner        FedoraSELinuxValidation.verify_selinux_info: PASSED   
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner test result summary
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner     TOTAL    : 2
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner     QUEUED   : 0
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner     ASSIGNED : 0
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner     RUNNING  : 0
2026-08-04 01:54:22.970[140166918605760][INFO] lisa.RootRunner     FAILED   : 0
2026-08-04 01:54:22.971[140166918605760][INFO] lisa.RootRunner     PASSED   : 2
2026-08-04 01:54:22.971[140166918605760][INFO] lisa.RootRunner     SKIPPED  : 0
2026-08-04 01:54:22.973[140166918605760][INFO] lisa.notifier[Html] report: /lisa/runtime/log/20260804/20260804-014713-049/lisa.html
2026-08-04 01:54:22.982[140166918605760][INFO] lisa. completed in 429.937 sec

### CODE REVIEW

**Comment by balakreddy** [comment]
Adding @jeremycline @nmeyerhans

File: lisa/microsoft/testsuites/fedora/selinux_validation.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_os=[Fedora]),
+        use_new_environment=True,
+    )
+    def verify_setenforce(self, node: Node) -> None:
+        self._assert_rpm_installed(node, ["libselinux", "libselinux-utils"])
+
+        # setenforce binary must exist and be executable
+        help_result = node.execute("setenforce --help", no_error_log=True)
```
Comment (jeremycline) [inline lisa/microsoft/testsuites/fedora/selinux_validation.py:114]: setenforce doesn't appear to have a --help flag and always just prints usage if the expected arg isn't provided. Other options would be `command --search setenforce` which will exist 0 and print the full path, or you could use `test -x /usr/bin/setenforce` which returns 0 if it exists and is executable, or you can use `which` as you do below with auditctl. Whatever direction you pick I think it should be consistent for both setenforce and auditctl

File: lisa/microsoft/testsuites/fedora/selinux_validation.py [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        Tests that getenforce accurately reports mode changes, sestatus
+        displays all required policy fields and contexts, and avcstat
+        shows live AVC cache statistics.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_os=[Fedora]),
+        use_new_environment=True,
+    )
+    def verify_selinux_info(self, node: Node) -> None:
+        self._assert_rpm_installed(node, ["policycoreutils"])
```
Comment (jeremycline) [inline lisa/microsoft/testsuites/fedora/selinux_validation.py:271]: IMO I don't think we should be asserting a package is installed, just that the commands you expect are there. Package names can change or commands can be shifted around within packages.

File: lisa/microsoft/testsuites/fedora/selinux_validation.py [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+            enf_check = node.execute("getenforce")
+            assert_that(enf_check.stdout.strip()).described_as(
+                "getenforce must report 'Enforcing' after setenforce 1"
+            ).is_equal_to("Enforcing")
+
+        finally:
+            restore_val = "1" if original_mode == "Enforcing" else "0"
+            node.execute(f"setenforce {restore_val}", sudo=True, no_error_log=True)
+
+        # === sestatus: standard fields ===
```
Comment (jeremycline) [inline lisa/microsoft/testsuites/fedora/selinux_validation.py:319]: I'm not sure we should be in the business of asserting a particular format to these various commands. It feels more likely to generate failures that aren't really "interesting". To me the things to test are:

1. Is SELinux enforcing by default?
2. Can it be toggled?
3. Are there any SELinux denials?

Separating 3 into its own test feels valuable since it's much more likely to fail.

**Comment by balakreddy** [comment]
Closing this pull request as the test are out of scope for the fedora image validation
